### PR TITLE
PYTHON-3093 Change streams support for user-facing PIT pre- and post-images

### PR DESCRIPTION
**Pull request by bcwarner**

### CODE REVIEW

File: test/change_streams/unified/change-streams-errors.json [hunk 1]
```diff
@@ -187,7 +187,7 @@
       "description": "change stream errors on ElectionInProgress",
       "runOnRequirements": [
         {
-          "minServerVersion": "4.4",
+          "minServerVersion": "4.2",
```
Comment (blink1073) [inline test/change_streams/unified/change-streams-errors.json:190]: We're pinning this to 4.4 pending https://jira.mongodb.org/projects/PYTHON/issues/PYTHON-3292

File: pymongo/mongo_client.py [hunk 2]
```diff
@@ -972,7 +973,8 @@ def watch(
             start_at_operation_time,
             session,
             start_after,
-            comment=comment,
+            comment,
+            full_document_before_change,
```
Comment (blink1073) [inline pymongo/mongo_client.py:977]: Needs an update to docstring for parameters and versionchanged

File: pymongo/database.py [hunk 3]
```diff
@@ -631,7 +636,8 @@ def watch(
             start_at_operation_time,
             session,
             start_after,
-            comment=comment,
+            comment,
+            full_document_before_change,
```
Comment (blink1073) [inline pymongo/database.py:640]: Needs an update to docstring for parameters and versionchanged

**Review by blink1073 — CHANGES_REQUESTED** [review]

**Comment by bcwarner** [comment]
Skips tests that weren't relevant.

**Review by juliusgeo — APPROVED** [review]
LGTM!

File: pymongo/database.py [hunk 4]
```diff
@@ -576,11 +579,13 @@ def watch(
             pipeline stages are valid after a ``$changeStream`` stage, see the
             MongoDB documentation on change streams for the supported stages.
           - `full_document` (optional): The fullDocument to pass as an option
-            to the ``$changeStream`` stage. Allowed values: 'updateLookup'.
+            to the ``$changeStream`` stage. Allowed values: 'updateLookup', 'whenAvailable', 'required'.
```
Comment (ShaneHarvey) [inline pymongo/database.py:582]: Does this line need to be applied to Database and MongoClient as well?

**Pull request merged**